tests/test_lag_2.py

```diff
@@ -0,0 +1,128 @@
+import pytest
+import time
+import logging
+from ptf_runner import ptf_runner
+from platform_fixtures import conn_graph_facts
+
+@pytest.fixture(scope="module")
+def common_setup_teardown(duthost, ptfhost, conn_graph_facts):
+    # TODO: get the testbed_type
+
+    lag_facts = duthost.lag_facts(host = duthost.hostname)['ansible_facts']
+    if lag_facts['names'] == []:
+        pytest.skip("No lag configuration found in %s" % duthost.hostname)
+
+    test_minlink = True
+    test_rate = True
+
+    # Add/update the public key
+    # TODO: convert add_container_to_inventory.yml
+
+    mg_facts = duthost.minigraph_facts(host=hostname)['ansible_facts']
+    fanout_neighbors = conn_graph_facts['device_conn']
+    vm_neighbors = mg_facts['minigraph_neighbors']
+
+    # Copy PTF test into PTF-docker for test LACP DU
+    test_files = ['lag_test.py', 'acs_base_test.py', 'router_utils.py']
+    for test_file in test_files:
+        src = "roles/test/files/acstests/%s" % test_file
+        dst = "/tmp/%s" % test_file
+        ptfhost.copy(src=src, dest=dst)
+    
+    # Copy tests to the PTF-docker
+    ptfhost.copy(src="ptftests", dest="/root")
+
+    # Inlucde testbed topology configuration
+    if testbed_type == 't1-lag':
+        # TODO
+    elif testbed_type == 't0':
+        # TODO
+    elif testbed_type == 't0-116'
+        # TODO
+
+    dut_mac = duthost['ansible_Ethernet0']['macaddress']
+
+    # Test each lag interface
+    for lag_name in lag_facts['names']
+        yield ptfhost, fanout_neighbors, vm_neighbors, mg_facts, lag_facts, lag_name
+
+def test_single_lag_lacp_rate(common_setup_teardown, testbed_devices):
+    ptfhost, fanout_neighbors, vm_neighbors, mg_facts, lag_facts, lag_name = common_setup_teardown
+
+    po = lag_name
+    po_interfaces = lag_facts[lag_name]['po_config']['ports']
+    po_intf_num = len(po_interfaces)
+    po_min_links = lag_facts[lag_name]['po_config']['runner']['min_ports']
+
+    # Pick flap interface name and calculate when it flaps, should portchannel interface flap or not
+    po_flap = float(po_intf_num - 1) / po_min_links * 100 < 75
+    flap_intf = lag_facts[lag_name]['po_config']['ports'].keys()[0]
+    if not po_flap:
```

Comment:
po_flap defined but never used in origin yml test,
I think it might use for logging? I add extra code to check it and will simple skip current test if the po_flap is false

Reply:
It is a copy-paste code and useless in current test case. I will remove it and rename flap_intf